### ISSUE TITLE
kernel errors

### ISSUE THREAD

**Issue by kupiqu** [blink1073/oct2py#148]
If I workaround issue #147 making octave-cli visible to the kernel with a soft link (/home/username/.local/bin/octave-cli), I get these errors:

```
Traceback (most recent call last):
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/spawnbase.py", line 166, in read_nonblocking
    s = os.read(self.child_fd, size)
OSError: [Errno 5] Input/output error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/expect.py", line 111, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/pty_spawn.py", line 459, in read_nonblocking
    incoming = super(spawn, self).read_nonblocking(size)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/spawnbase.py", line 171, in read_nonblocking
    raise EOF('End Of File (EOF). Exception style platform.')
pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "pyOct2PyTest.py", line 5, in <module>
    from oct2py import Oct2Py
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/oct2py/__init__.py", line 38, in <module>
    octave = Oct2Py()
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/oct2py/core.py", line 82, in __init__
    self.restart()
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/oct2py/core.py", line 528, in restart
    logger=self.logger)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/octave_kernel/kernel.py", line 162, in __init__
    self.repl = self._create_repl()
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/octave_kernel/kernel.py", line 374, in _create_repl
    force_prompt_on_continuation=True)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/metakernel/replwrap.py", line 97, in __init__
    continuation_prompt_regex))
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/metakernel/replwrap.py", line 118, in set_prompt
    self.child.expect(prompt_regex)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/spawnbase.py", line 341, in expect
    timeout, searchwindowsize, async_)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/spawnbase.py", line 369, in expect_list
    return exp.expect_loop(timeout)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/expect.py", line 117, in expect_loop
    return self.eof(e)
  File "/home/miniconda/envs/cenv1/lib/python3.7/site-packages/pexpect/expect.py", line 63, in eof
    raise EOF(msg)
pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
<pexpect.pty_spawn.spawn object at 0x7f389d72f400>
command: /home/username/.local/bin/octave-cli
args: [b'/home/username/.local/bin/octave-cli', b'--interactive', b'--quiet', b'--no-init-file']
buffer (last 100 chars): ''
before (last 100 chars): 'ing shared libraries: liboctinterp.so.7: cannot open shared object file: No such file or directory\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 127
flag_eof: True
pid: 21260
child_fd: 7
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('octave.*>')
```

**Comment by blink1073** [comment]
Are you trying to soft-link to an `octave-cli` or `octave` itself?  We only support the `octave-cli` executable.

**Comment by kupiqu** [comment]
I soft-linked to octave-cli, but octave 5 is smart enough so it called from terminal it behaves as octave-cli. anyways, this discussion should be in #147 not here, this is a different bug, once octave-cli is set by hand.

**Comment by kupiqu** [comment]
by set by hand I mean soft-linked instead of a path added in oct2py call

the error here (errors after finding octave-cli) and that of #147 (not finding octave-cli when referenced in oct2py call) are not the same, so one cannot be duplicate of the other

**Comment by kupiqu** [comment]
Ok, it seems there is an error in how Octave 5 releases octave-cli, as it is not functional, at least when packaged in flatpak (will check out with them).

I workaround the issue creating a soft-link to octave instead of octave-cli, because as I was saying octave behaves as octave-cli when called from terminal.

This fixes this issue, so closing, however, I still think issue #147 needs fixing.
